src/shell.js

```diff
@@ -229,7 +229,7 @@ if (ENVIRONMENT_IS_NODE) {
 #endif
 
   quit_ = (status, toThrow) => {
-    if (keepRuntimeAlive()) {
+    if (runtimeKeepaliveCounter) {
```

Comment:
I think the old code is more technically correct...  the reason its an issue for d8 is that d8 doesn't have a way to naturally exit like node does.. which means the EXIT_RUNTIME=0 (the default) will always stay alive forever .. which is not what the unit tests want.

I think its ok for node and d8 to differ here.. please a comment below though.

Alternatively we could force EXIT_RUNTIME=1 whenever we build with `shell` in `ENVIRONMENT`?  Or for all our unit tests?

Reply:
what do you mean by "naturally exit"? I think node `process.exit` is similar to d8 `quit`, no? Both halt immediately with an exit code.

The only difference I see is that node also has process.exitCode which we can set ahead of time, and I guess it applies even if we don't exit using `process.exit`?

But maybe I'm missing something here...

Reply:
I mean fall of the end of the event loop when there is nothing more to do. 

Calling `process.exit` is pretty aggressive thing to do since it brings down everything running under node.. not just the emscripten module (imagine using an emscripten module on the server in node for example).  The less we can rely on that the better.

Reply:
Its also nicer to throw be somebody can catch the exception.

Reply:
Ah, then the difference is that in node we can set `process.exitCode` for when we fall out of the end of the program? I can add that in a comment then.